Improve Readme (#46)
celo-org/rosetta: README.md

# Celo Rosetta

[![CircleCI](https://circleci.com/gh/celo-org/rosetta/tree/master.svg?style=shield)](https://circleci.com/gh/celo-org/rosetta/tree/master)
[![License](https://img.shields.io/github/license/celo-org/rosetta.svg)](https://github.com/celo-org/rosetta/blob/master/LICENSE.txt)

A monitoring server for celo-blockchain

> ## What is Celo-Rosetta?

Reviewer: ```suggestion
## What is Celo Rosetta?
```

> Celo Rosetta is an rpc server that exposes an API to:

Reviewer: ```suggestion
Celo Rosetta is an RPC server that exposes an API to:
```

>  * Query Blockchain

Reviewer: ```suggestion
 * Query Celo's Blockchain
```

 * Obtain Balance Changing Operations

>  * Construct Airgapped transactions

Reviewer: ```suggestion
 * Construct Airgapped Transactions
```

> With a special focus on getting balanche change operations, provides an easy way to obtain changes that are not easily queryable using

Reviewer: ```suggestion
With a special focus on getting balance change operations, Celo Rosetta provides an easy way to obtain changes that are not easily queryable using
```

the celo-blockchain rpc; such as:

* Gas Fee distribution
* Gold transfers (internal & external). Taking in account Tobin Tax
* Epoch Rewards Distribution
* LockedGold & Election Operations

## RPC endpoints

> Rosetta exposes the following endpoints:

Reviewer: Is Rosetta == Celo Rosetta?

Reviewer: Yeah... agree, we should ask Patrick what the terminology should be for the implementations, eg. `Celo Rosetta Interface`?

Author: he said rosetta it's fine, since it's within `celo-org` namespace.

I'll be using 'celo rosetta' in the readme

 * `POST /mempool/transaction`: Get a Mempool Transaction
 * `POST /account/balance`: Get an Account Balance
 * `POST /construction/metadata`: Get Transaction Construction Metadata
 * `POST /construction/submit`: Submit a Signed Transaction

For an understanding of inputs & outputs check [servicer.go](./service/rpc/servicer.go)

## Running Rosetta Docker Image

Rosetta is released as a docker image: `us.gcr.io/celo-testnet/rosetta`. All version can be found on the [registry page](https://us.gcr.io/celo-testnet/rosetta)

Within the docker image, we pack `rosetta` binary and also `geth` binary from celo-blockchain. Rosetta will run both.

### Configuration

The necessary parameters to run rosetta are:

 * `genesis.json` for the target network (can be found by `curl 'https://storage.googleapis.com/genesis_blocks/baklava' > genesis.json`)
 * `staticNode` enodeURL. Rosetta will directly peer to the list of staticNode provided (it doens't run discovery protocol for the moment).
    This node can be any you have access to. For a public list check `https://storage.cloud.google.com/static_nodes/baklava`

Additionaly, it needs a data directory for the geth datadir & rosetta.db

### Run

> To run rosetta:

Reviewer: sometimes Rosetta is caps-cased, other times not -- let's choose one and stick w it?

  validator-data/bootstrap_balances.json
rosetta-validator check:complete
```

### Running on development

> #### Running on RC1:

Reviewer: should we move RC1 out of development section?

Author: why?
It's on this section because it uses `go run main.go` which is what you during development.

Reviewer: oh my bad.. I got confused since I confused dev network with development process, since it said `Running on development`. Maybe the title can be changed to `Developing Locally with Celo Rosetta Interface` or something?

Author: mmm.. no need imo, it's already within `Developer Guide`
